wasm/samples/fees_executor/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+//! Iroha executor with fees support.
+
+#![no_std]
+
+#[cfg(not(test))]
+extern crate panic_halt;
+
+extern crate alloc;
+use alloc::format;
+use iroha_executor::{
+    data_model::{isi::CustomInstruction, query::builder::SingleQueryError, parameter::{CustomParameter, Parameter}},
+    prelude::*, permission::ExecutorPermission as _,
+};
+use fees_executor_data_model::{parameters::*};
+use iroha_schema::IntoSchema;
+use serde::{Deserialize, Serialize};
+use iroha_executor_data_model::parameter::Parameter as ExecutorParameter;
+
+use dlmalloc::GlobalDlmalloc;
+use iroha_executor::{data_model::block::BlockHeader, prelude::*};
+
+#[global_allocator]
+static ALLOC: GlobalDlmalloc = GlobalDlmalloc;
+
+#[derive(Debug, Clone, Visit, Execute, Entrypoints)]
+#[visit(custom(
+    visit_set_parameter,
+))]
+struct Executor {
+    host: Iroha,
+    context: Context,
+    verdict: Result,
+}
+
+fn find_fee_options(executor: &mut Executor) -> FeesOptions {
+    let parameters = executor.host().query_single(FindParameters).dbg_unwrap();
+
+    let fees_options: FeesOptions = parameters
+        .custom()
+        .get(&FeesOptions::id())
+        .unwrap()
+        .try_into()
+        .expect("INTERNAL BUG: Failed to deserialize json as `FeesOptions`");
+
+    fees_options
+}
+
+fn visit_set_parameter(executor: &mut Executor, isi: &SetParameter) {
+    if let Parameter::Custom(param) = isi.parameter() {
+        if param.id() == &FeesOptions::id() {
+            let payload = param.payload();
+            match payload.try_into_any::<FeesOptions>() {
+                Ok(options) => {
```

Comment:
Not sure what you can do about it but I think you should also do a check of relationship between the account and assetDefinitionId

Reply:
Made fees options read-only for the moment. I expect there should be isi to set them.